IOPool/Streamer/src/StreamerInputFile.cc

```diff
@@ -40,21 +43,26 @@ namespace edm {
     readStartMessage();
   }
 
-  StreamerInputFile::StreamerInputFile(std::string const& name, std::shared_ptr<EventSkipperByID> eventSkipperByID)
-      : StreamerInputFile(name, name, eventSkipperByID) {}
+  StreamerInputFile::StreamerInputFile(std::string const& name,
+                                       std::shared_ptr<EventSkipperByID> eventSkipperByID,
+                                       unsigned int prefetchMBytes)
+      : StreamerInputFile(name, name, eventSkipperByID, prefetchMBytes) {}
 
   StreamerInputFile::StreamerInputFile(std::vector<FileCatalogItem> const& names,
-                                       std::shared_ptr<EventSkipperByID> eventSkipperByID)
+                                       std::shared_ptr<EventSkipperByID> eventSkipperByID,
+                                       unsigned int prefetchMBytes)
       : startMsg_(),
         currentEvMsg_(),
         headerBuf_(1000 * 1000),
         eventBuf_(1000 * 1000 * 7),
+        tempBuf_(1024 * 1024 * prefetchMBytes),
         currentFile_(0),
         streamerNames_(names),
         multiStreams_(true),
         currentFileName_(),
         currentFileOpen_(false),
         eventSkipperByID_(eventSkipperByID),
+        prefetchMBytes_(prefetchMBytes),
```

Comment:
It seems to me the `prefetchMbytes_` has now become mostly unnecessary. The check
https://github.com/cms-sw/cmssw/blob/3db4eedaf79dafc7d94272ac61199ee4c329044f/IOPool/Streamer/src/StreamerInputFile.cc#L129
could be replaced with
```cpp
if (not tempBuf_.empty()) {
```

Reply:
Good point. I removed prefetchMBytes_.